clang/include/clang/Sema/Sema.h

```diff
@@ -1967,6 +1973,11 @@ class Sema final : public SemaBase {
   bool CheckFunctionCall(FunctionDecl *FDecl, CallExpr *TheCall,
                          const FunctionProtoType *Proto);
 
+  bool BuiltinVectorMath(CallExpr *TheCall, QualType &Res);
+  bool BuiltinVectorToScalarMath(CallExpr *TheCall);
+
+  bool CheckHLSLBuiltinFunctionCall(unsigned BuiltinID, CallExpr *TheCall);
```

Comment:
Shouldn't that be moved to SemaHLSL?

Reply:
I tried, but it depends on a lot of static functions inside `SemaChecking.cpp`, so I decided to leave it for later.